text/0096-auto-batching.md

```diff
@@ -0,0 +1,86 @@
+- Title: Auto-Batching
+
+# Auto-Batching
+
+## 1. Summary
+
+Meilisearch can automatically group consecutive asynchronous `documentAddition` or `documentPartial` tasks for the same index via an automatic batching mechanism.
+
+The user can enable this auto-batching behavior through various command flag options.
+
+## 2. Motivation
+
+We have regularly collected user pain points pointing out the slow indexing over the last year. We explained several times to users to make batches containing a maximum of documents to be updated/added to compress the indexing time of specific data structures.
+
+To make Meilisearch easier to use, we explored the idea of automatically creating these batches within Meilisearch before indexing users’ documents.
+
+## 3. Functional  Specification
+
+### 3.1. Explanations
+
+All `tasks` are linked to a `batchUid` field. The `batchUid` field identifies that several identical, consecutive tasks have been grouped in the same batch.
+
+Only consecutive `documentAddition` and `documentPartial` tasks for the same index can have the same `batchUid`. All `tasks` concerning other operations will also be part of a `batchUid` having only one task.
+
+#### 3.1.1. Grouping tasks to a single batch
+
+The scheduling program that groups tasks within a single batch is triggered when an asynchronous `task` currently processed reaches a terminal state as `succeeded` or `failed`.
+
+In other words, when a scheduled `task` is picked from the FIFO task queue, the scheduler fetches and groups all consecutive `documentAddition` for a similar index in a batch until it encounters another task type or a similar task type but for a different index.
```

Comment:
Actually, it doesn't stop on the first update of another index. It takes as many updates as possible for an index, possibly skipping over updates for other indexes to take more.

Reply:
Was this added late? I am pretty sure that the original behavior was as described in the specification 🤯

I'll describe it.

The review is late though. It's probably going to be a last-minute job for the documentation team and dev rel team that is already well advanced for the v026.0 release. 😟